### PR TITLE
HeatPump:PlantLoop:EIR:*: new fields

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

- [x] Support new fields for `HeatPump:PlantLoop:EIR:Cooling`:
  - "Control Type"
  - "Flow Mode"
  - "Minimum Part Load Ratio"
  - "Minimum Source Inlet Temperature"
  - "Maximum Source Inlet Temperature"
  - "Minimum Supply Water Temperature Curve Name"
  - "Maximum Supply Water Temperature Curve Name"
- [x] Support new fields for `HeatPump:PlantLoop:EIR:Heating`:
  - "Heating To Cooling Capacity Sizing Ratio"
  - "Heat Pump Sizing Method"
  - "Control Type"
  - "Flow Mode"
  - "Minimum Part Load Ratio"
  - "Minimum Source Inlet Temperature"
  - "Maximum Source Inlet Temperature"
  - "Minimum Supply Water Temperature Curve Name"
  - "Maximum Supply Water Temperature Curve Name"
  - "Dry Outdoor Correction Factor Curve Name"
  - "Maximum Outdoor Dry Bulb Temperature For Defrost Operation"
  - "Heat Pump Defrost Control"
  - "Heat Pump Defrost Time Period Fraction"
  - "Defrost Energy Input Ratio Function of Temperature Curve Name"
  - "Timed Empirical Defrost Frequency Curve Name"
  - "Timed Empirical Defrost Heat Load Penalty Curve Name"
  - "Timed Empirical Defrost Heat Input Energy Fraction Curve Name"

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/model/HeatPumpPlantLoopEIRCooling.cpp [hunk 1]
```diff
@@ -390,6 +462,11 @@ namespace model {
 
     setReferenceCoefficientofPerformance(7.5);  // IDD default
```
Comment (joseph-robertson) [inline src/model/HeatPumpPlantLoopEIRCooling.cpp:463]: This IDD default has changed to 3.0. Should that be captured here?

File: src/model/HeatPumpPlantLoopEIRCooling.cpp [hunk 2]
```diff
@@ -450,6 +527,11 @@ namespace model {
 
     setReferenceCoefficientofPerformance(7.5);
```
Comment (joseph-robertson) [inline src/model/HeatPumpPlantLoopEIRCooling.cpp:528]: Here too.

**Review by jmarrec — CHANGES_REQUESTED** [review]
Model tests, FT tests, required-field, all good.

Missing a min fields update, and some resetXXXCurve function and this is good to go!

**Comment by joseph-robertson** [comment]
@jmarrec segfault in tests - presumably related to new reset curve methods. do you see anything obvious?

File: src/model/HeatPumpPlantLoopEIRHeating.cpp [hunk 3]
```diff
@@ -875,14 +910,26 @@ namespace model {
     return getImpl<detail::HeatPumpPlantLoopEIRHeating_Impl>()->setMinimumSupplyWaterTemperatureCurve(minimumSupplyWaterTemperatureCurve);
   }
 
+  void HeatPumpPlantLoopEIRHeating::resetMinimumSupplyWaterTemperatureCurve() {
+    getImpl<detail::HeatPumpPlantLoopEIRCooling_Impl>()->resetMinimumSupplyWaterTemperatureCurve();
+  }
```
Comment (jmarrec) [inline src/model/HeatPumpPlantLoopEIRHeating.cpp]: Segfault is here @joseph-robertson

**Comment by jmarrec** [comment]
We forgot VT! cf https://github.com/NREL/OpenStudio/issues/4977

**Pull request merged**